### ISSUE TITLE
Exception diaper just raises

### ISSUE THREAD

**Issue by donspaulding** [MindscapeHQ/raygun4py#38]
Hi there,

Looking at the new release, I see that the send_exception() function on the raygunprovider.RaygunSender class attempts to delete some of its own local variables.

https://github.com/MindscapeHQ/raygun4py/blob/master/python2/raygun4py/raygunprovider.py#L99

To my eye, the catching and re-raising of that exception accomplishes nothing.  Is it there for a reason?

**Comment by fundead** [comment]
The reason for this is to prevent a circular reference which the GC cannot handle efficiently. Please see this issue for more details: https://github.com/MindscapeHQ/raygun4py/issues/26 (and the official documentation of this behavior in the warning at https://docs.python.org/2/library/sys.html#sys.exc_info).

I'll add a code comment for those lines in the next release as it does look odd if the reader is unfamiliar with that behavior (it is rather counter-intuitive).

**Comment by donspaulding** [comment]
I think I understand what the code is trying to accomplish, but I still think it's broken.  As I understand the docs you linked to, the recommended pattern for avoiding the circular reference is more something like this:

```python
def some_exception_handler():
    exc_type, exc_val, exc_tb = sys.exc_info()
    try:
        do_stuff_with_exc_info(exc_type, exc_val, exc_tb)
    finally:
        del exc_type, exc_val, exc_tb
```
The biggest distinction of note is that `del`ing the references to the exc_info() values is not what needs to be guarded by the try clause, it's the cleanup action that needs to take place should anything you are doing with the exc_info values in the try clause throw any kind of exception.

`del some_var` doesn't throw exceptions when the name `some_var` is in scope, which it plainly is with just a quick glance at the code, so you don't need to wrap that in a try clause.  As it sits now, the only way your except clause will be triggered is via some bug in the python interpreter.  Not a very likely scenario.

Also, even if you were concerned about `del exc_type, exc_value, exc_traceback` somehow failing in the try clause, the following two pieces of code are equivalent in python:

```python
try:
    do_something()
except Exception:
    raise
```
and
```python
do_something()
```
In other words, if you have nothing but `raise` in your except clause, you might as well not wrap the try clause at all, because you're throwing the error up to your caller regardless.

**Comment by fundead** [comment]
Hi Don,

I'm inclined to agree with the analysis above, had a quick attempt during the last release at reproducing the actual behavior which didn't work. In order to confirm the impact of this change it looks like I'll need to create a proper repro and profile the GC to see the circular reference. This will allow us to verify removing the raise/altering the logic. As the code is running in production with no (reported) issues as this time we have a policy on not making breaking changes without evidence, so I'll be revisiting this issue and hopefully fixing the logic once I have time to create the repro.
